integration-tests/pkg/assert/assert.go

```diff
@@ -0,0 +1,44 @@
+package assert
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/stretchr/testify/assert"
+)
+
+func ElementsMatchFunc[N any](expected []N, actual []N, equal func(a, b N) bool) bool {
+	if len(expected) != len(actual) {
+		return false
+	}
+
+	for i := range expected {
+		if !equal(expected[i], actual[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+func ListsToAssertMsg[N any](expected []N, actual []N) string {
```

Comment:
I don't think this needs to be exposed outside this package - all message handling should happen internally